scilpy/tractograms/streamline_and_mask_operations.py

```diff
@@ -174,74 +360,95 @@ def cut_between_mask_two_blobs_streamlines(sft, binary_mask_1,
+    idx, streamline, pts_to_idx, roi_data_1, roi_data_2
+):
     """
-    Cut streamlines so their segment are going from binary mask #1 to binary
+    Cut streamlines so their segment are going from label mask #1 to label
     mask #2. New endpoints may be generated to maximize the streamline length
     within the masks.
-    """
-    new_streamlines = []
-    kept_idx = []
-    # Get the indices of the "voxels" intersected by the streamlines and the
-    # mapping from points to indices.
-    (indices, points_to_idx) = uncompress(streamlines, return_mapping=True)
 
-    if len(streamlines[0]) != len(points_to_idx[0]):
-        raise ValueError("Error in the uncompress function. Try running the "
-                         "scil_tractogram_remove_invalid.py script with the \n"
-                         "--remove_single_point and --remove_overlapping_points"
-                         " options.")
-
-    for strl_idx, strl in enumerate(streamlines):
-        # The "voxels" intersected by the current streamline
-        strl_indices = indices[strl_idx]
-        # Find the first and last "voxels" of the streamline that are in the
-        # ROIs
-        in_strl_idx, out_strl_idx = _intersects_two_rois(roi_data_1,
-                                                         roi_data_2,
-                                                         strl_indices)
-        # If the streamline intersects both ROIs
-        if in_strl_idx is not None and out_strl_idx is not None:
-            points_to_indices = points_to_idx[strl_idx]
-            # Compute the new streamline by keeping only the segment between
-            # the two ROIs
-            cut_strl = compute_streamline_segment(strl, strl_indices,
-                                                  in_strl_idx, out_strl_idx,
-                                                  points_to_indices)
-            # Add the new streamline to the sft
-            new_streamlines.append(cut_strl)
-            kept_idx.append(strl_idx)
-
-    return new_streamlines, kept_idx
+    Parameters
+    ----------
+    idx: np.ndarray
+        Indices of the voxels intersected by the streamlines.
+    streamline: np.ndarray
+        The streamlines to cut.
+    pts_to_idx: np.ndarray
+        Mapping from points to indices.
+    roi_data_1: np.ndarray
+        Boolean array representing the region #1.
+    roi_data_2: np.ndarray
+        Boolean array representing the region #2.
+
+    Returns
+    -------
+    new_strmls : list of np.ndarray
+        New streamlines trimmed within the masks.
+    """
+    # Find the first and last "voxels" of the streamline that are in the
+    # ROIs
+    in_strl_idx, out_strl_idx = _intersects_two_rois(roi_data_1,
+                                                     roi_data_2,
+                                                     idx)
+
+    cut_strl = []
+    # If the streamline intersects both ROIs
+    if in_strl_idx is not None and out_strl_idx is not None:
```

Comment:
To make it work with my data (Whole brain with 6 spheres not touching being 6 labels):

In the if just return cut_strl and add a else: to return None